plugins/modules/ndo_pod_settings.py

```diff
@@ -0,0 +1,304 @@
+        set_pod_settings_policy_names(mso_template, match.details)
+        mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = [set_pod_settings_policy_names(mso_template, obj) for obj in match]  # Query all objects
+
+    pod_settings_path = "/fabricPolicyTemplate/template/podPolicyGroups/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+
+        mso_values = {
+            "name": name,
+            "description": description,
+        }
+
+        if macsec_policy or macsec_policy_uuid:
+            mso_values["fabricMACsec"] = mso_template.get_macsec_policy_object(macsec_policy_uuid, macsec_policy, fail_module=True).details.get("uuid")
+        if ntp_policy or ntp_policy_uuid:
+            mso_values["ntp"] = mso_template.get_ntp_policy_object(ntp_policy_uuid, ntp_policy, fail_module=True).details.get("uuid")
+
+        if match:
+            remove_data = []
+            unwanted = []
+            if macsec_policy == "" or macsec_policy_uuid == "":
+                remove_data.append("fabricMACsec")
+                unwanted.extend(["fabricMACsec", "macsec_policy"])
+            if ntp_policy == "" or ntp_policy_uuid == "":
+                remove_data.append("ntp")
+                unwanted.extend(["ntp", "ntp_policy"])
+
+            append_update_ops_data(ops, match.details, pod_settings_path, mso_values, remove_data)
+            mso.sanitize(mso_values, collate=True, unwanted=unwanted)
+            set_pod_settings_policy_names(mso_template, mso.proposed)
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=pod_settings_path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=pod_settings_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = mso_template.get_pod_settings_object(uuid, name, search_object=response)
+        if match:
+            set_pod_settings_policy_names(mso_template, match.details)
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        set_pod_settings_policy_names(mso_template, mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+    mso.exit_json()
+
+
+def set_pod_settings_policy_names(mso_template, pod_settings):
+    if pod_settings.get("fabricMACsec"):
+        pod_settings["macsec_policy"] = mso_template.get_macsec_policy_object(uuid=pod_settings["fabricMACsec"]).details.get("name")
+    if pod_settings.get("ntp"):
+        pod_settings["ntp_policy"] = mso_template.get_ntp_policy_object(uuid=pod_settings["ntp"]).details.get("name")
+    return pod_settings
```

Comment:
```suggestion
    if pod_settings.get("fabricMACsec"):
        pod_settings["fabricMACsecName"] = mso_template.get_macsec_policy_object(uuid=pod_settings["fabricMACsec"]).details.get("name")
    if pod_settings.get("ntp"):
        pod_settings["ntpName"] = mso_template.get_ntp_policy_object(uuid=pod_settings["ntp"]).details.get("name")
    return pod_settings
```
To be consistent with other modules.

Reply:
not sure I agree with the choice here of using a non-existing naming but made the change for consistency purposes